Source/Tools/LinuxEmulation/LinuxSyscalls/x64/Semaphore.cpp

```diff
@@ -29,8 +29,10 @@ void RegisterSemaphore(FEX::HLE::SyscallHandler* Handler) {
     case IPC_SET: {
       struct semid64_ds buf {};
       buf = *semun.buf;
+      FaultSafeUserMemAccess::VerifyIsReadable(semun.buf, sizeof(*semun.buf));
```

Comment:
Check should be moved before the previous line.

Reply:
Moved before